[IDB IGNORE][MDB IGNORE] Adds Jungleland as a randomly chosen alternative to Lavaland, [Bounty]

Djiq opened the pull request:
# Document the changes in your pull request
This PR adds an alternative to Lavaland that is themed after a toxic jungle on a planet brimming with sulphur and acidic pits, a small island filled with life has been found, it is insanely rich in resources, and so NT planted a base there, but the aggressive jungle fauna will not be easily defeated!

Ok this is a massive really - fucking - massive PR. First of all, Jungleland is divided into 5 Biomes

Deep Jungle - standard jungle experience, i believe it is the easiest part of the jungle as it only contains dryads and blobbies
Dying Jungle - a part of jungle that is withering away for *one* reason or another, it is a harder area, as it contains more dangerous mobs like corrupted dryads or magmawings.
Toxic Pit - A swamp filled with sulphuric water that you can easily get poisoned by, this area is quite brutal as it often spawns Medurachas and a lot of mosquitos
Rocky Beach - a transitory zone between Toxic Pits and Barren Rocks, some enemies from both
Barren Rocks - Lavaland like biomes containing watchers and goliaths.

This PR also adds unique ruins made by @Marmio64, you should be able to check them out in mapdiff, i dont really have a way to describe them, just some thematic ruins for most biomes.

Besides that this PR also aims to add a couple new vegatation and chemicals that you can gather from Jungle flora (Currently only Polybycin and Quinine are implemented, but im planning on adding more during the TM phase), some more items related to Jungleland mobs (currently the mobs unique to jungleland dont drop anything, but that is soon to be changed), and it also adds a new unique disease - Malaria Exotica, you can get it from getting bitten by Mosquitos and it is a slowly ramping up disease with periods of asymptomacity that after a couple minutes kills you by extreme fever. 

# Roadmap

- [x] Add ruins spawning
- [x] Add mining base
- [x] Add mob loot
- [x] Add megafauna spawning
- [x] Implement a megafauna unique to jungleland
- [x] Add unique vegetation along with the rest of chemicals
- [x] Add a way for admins to force either lavaland or jungleland to the following round

# Examples of Jungleland world gen
![obraz](https://user-images.githubusercontent.com/42111655/178273300-145dc5bd-6857-4117-b742-af37b8ff5bf3.png)
![obraz](https://user-images.githubusercontent.com/42111655/178273316-2bca68bf-b32c-4767-b8ee-449091caf40b.png)
![obraz](https://user-images.githubusercontent.com/42111655/178273329-8b5b1390-4e72-4ece-aa1c-d29fa3810469.png)


# Spriting
Once IDB issues are fixed, please look into that check to see all of the new sprites, there are literally to many to list here.

# Wiki Documentation
waiting for my work to be ported to a proper page, here is how it looks for now:
https://wiki.yogstation.net/wiki/User:Djiq

# Changelog
:cl:  
rscadd: New Mining area: Jungleland, Most content added by Djiq, Most ruins, some mobs and Ivymen added by Marmio64. Other ruins added by ToasterBiome. Various fixes and updates by courtesy of Jamie1D and SapphicOverload!
rscadd: New Mobs now appear on jungleland, such as Meduracha, Dryads, Giant mosquitoes, and Skin Twisters!
rscadd: New Mining implement: Kinetic Javelin, get it either from the roundstart kit, or buy it in a mining vendor.
rscadd: New Megafauna: Tar King - You can summon it by collecting 3 parts of a broken crystal, assembling it, and placing it in a Forgotten Altar!
rscadd: Ivymen - Jungleland's counterpart to ashwalkers, by Marmio64
rscadd: Jungleland has many new biomes, beware of sulphuric pits, as it will cause you trouble if you step into it.
imageadd: Added new icons and icon states for jungleland
mapping: Added Jungleland as a new mining destination!
/:cl:

Marmio64 committed:
adds a guthen to ivymen nest

Djiq committed:
Merge pull request #26 from Marmio64/ivymen-nest-guthen

adds a guthen to ivymen nest

Marmio64 committed:
updates some ruin appearances

Marmio64 committed:
more ruin changes

Marmio64 committed:
slightly lowered mob density, slightly higher chance of emerald spiders

Marmio64 committed:
adds the standard marrow weavers

Djiq committed:
Merge pull request #28 from Marmio64/mob-density

slightly lower mob density, slightly higher chance of emerald spiders

Djiq committed:
Merge pull request #27 from Marmio64/improve-the-ruins

Improves the look of some of the ruins

Marmio64 committed:
fixes rock turfs, expands xeno ruin

Djiq committed:
Merge pull request #30 from Marmio64/better-xeno-ruin

expands and improves xeno ruin

Marmio64 committed:
couple of loot changes

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
sprotes

Djiq committed:
Merge pull request #31 from Marmio64/morelootchanges

couple of loot changes

Djiq committed:
changes obj/screen to atom/mobable/screen because reasons

Djiq committed:
makes it compile and changes how dilithium and bs generates a bit

Djiq committed:
addresses some common concerns

Djiq committed:
more common concerns

JamieD1 committed:
Update JungleBiomes.dm

Marmio64 committed:
ivymen egg balancing

Djiq committed:
Merge pull request #32 from Marmio64/ivymen-egg-balance

ivymen egg balancing

Djiq committed:
doing this again

Djiq committed:
Merge remote-tracking branch 'origin/jungleland' into jungleland

Marmio64 committed:
carp pond ruin + ruin descriptions

Djiq committed:
Merge pull request #33 from Marmio64/carppond-and-ruin-descriptions

carp pond ruin + ruin descriptions

Djiq committed:
crushies

Djiq committed:
Merge branch 'jungleland'

Djiq committed:
icons

Djiq committed:
rustg all-features

Djiq committed:
tonight im feeling fruity

Djiq committed:
alpha meduracha

Djiq committed:
alpha meduracha 2

Djiq committed:
alpha meduracha finished

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
blobby alpha

Djiq committed:
alpha blobby, 3 kinetic crusher trophies done, changes to jungle nests

Djiq committed:
makes satchels work on jungleland

Djiq committed:
e

Djiq committed:
lintr

Djiq committed:
fixes mining base atmos filters, QOL changes like giga drill and charges working

Djiq committed:
more fixies

Djiq committed:
beginnings of obsidian pillars

Djiq committed:
Merge remote-tracking branch 'upstream/master' into jungleland

Djiq committed:
makes sure it actually compiles

Djiq committed:
Merge remote-tracking branch 'upstream/master' into jungleland

Djiq committed:
adds more sand to jungleland and fixes tar mobs not working with kinetic crusher

Djiq committed:
changies

patpol4 reviewed (dismissed):
I'll do some now and some later

1 jungleland_dead_bubblegum has some duplicate blood decals, is this intentional?
![image](https://user-images.githubusercontent.com/48154165/215522725-3ee51427-0a7f-475b-b623-337c4ae78548.png)
2. jungleland_dead_tartemple Replace this turf with sand, it will blend better
![image](https://user-images.githubusercontent.com/48154165/215524729-df38e7be-bc77-4863-a7f1-d690b2e7de15.png)

Actually during the making of this review i realized it would be easier if i just did changes to the maps to your branch directly, is that ok? There's plenty of small changes i would  like do to make it look and feel better.

Djiq committed:
Merge branch 'master' into jungleland

Djiq committed:
i think i can slowly get back to working on this

Djiq committed:
Merge remote-tracking branch 'origin/jungleland' into jungleland

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
stuff

Djiq committed:
more stuff

Djiq committed:
more alpha mobbs

Djiq committed:
all crusher trophies finished

Djiq committed:
new miner weapon coz why not

Djiq committed:
e

Djiq committed:
basics for a tar assistant

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
i cant find SPECIES_PERK_ICON folder anywhere, i feel slow drain on my sanity as this issue has already consumed 2 hours of my time

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
updaties

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
thingies

Djiq committed:
TM ready

Djiq committed:
PROC_REF (TM ready i promise)

Djiq committed:
guh

Djiq committed:
TM fixes pt1

Djiq committed:
more fixes

Djiq committed:
TM bug fix

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
more stuff

Djiq committed:
kwarry nerf

Djiq committed:
e

Marmio64 committed:
3 new ruins + less punji sticks

Marmio64 committed:
moves cave to swamp and adds loot

Marmio64 committed:
forgot disposals pipe for seed vault

Djiq committed:
Merge pull request #35 from Marmio64/newruin

3 new ruins and less punji sticks

Djiq committed:
javelin fixes

Djiq committed:
reduces lag hopefully

Marmio64 committed:
fixivymenstuff

Djiq committed:
more fixes

Djiq committed:
Merge pull request #36 from Marmio64/fixivymen

fix some ivymen stuff

Djiq committed:
misc fixes

Djiq committed:
fixes skin twisters deleting people, for fucks sake who changes weakref behaviour to that????

Djiq committed:
Merge remote-tracking branch 'origin/jungleland' into jungleland

Djiq committed:
Merge branch 'master' into jungleland

Djiq committed:
reverts AAC fix

SapphicOverload committed:
toxic buildup fixes

SapphicOverload committed:
bit too much healing for toxin lovers

Djiq committed:
adjusts mosquitos

Djiq committed:
Merge pull request #37 from SapphicOverload/toxic-buildup-fixes

Toxic buildup fixes

Marmio64 committed:
gems, loot changes, and electrified core nerf

electrified javelin core damage reduced to 15 from 35

removes bones from dryad (its a plant spirit)

mosquito chitin reduced to 1 from 4 (they're not hard to kill)

yellowjacket chitin reduced from 4 to 2, and sinew from 2 to 1.

all alpha mobs get their original monster's butcher results about times 2 (if they didnt have it already), and a 2 low level mining gem (worth 200 mining points each)

tar king drops a primal tarstone gem, which can be claimed and sold on the cargo shuttle for 3000 mining points.

and some spelling fixes

Marmio64 committed:
1 extra bone from emerald spider

Marmio64 committed:
yellowjacket drops one bone

this doesnt make much more sense than dryads but there does need to be more sources of bones

Djiq committed:
Merge pull request #38 from Marmio64/core+lootchange

gems, loot changes, and electrified core nerf

Djiq committed:
Fixes ITEM_SLOT_NECK and ITEM_SLOT_HEAD

Djiq committed:
remove debug info i accidentally left in

Marmio64 committed:
improvements and stuff

explosive shrooms are less common

dying jungle less cluttered

punji sticks in jungle tar temple moved inside so if you come in, look at them, and then step on them, its completely your fault.

also slightly improves the look of the swamp cave ruin

Djiq committed:
Merge pull request #39 from Marmio64/calmtheminers

improvements and stuff

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
Merge remote-tracking branch 'origin/jungleland' into jungleland

Djiq committed:
maintenance

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
obsidian

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
e

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
e

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
Merge branch 'master' into jungleland

Djiq committed:
up to date and funky stuff

Djiq committed:
jg jgjgjg

Djiq committed:
fixies

Djiq committed:
more fixies

Djiq committed:
shielpep

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
e

Djiq committed:
this is the begginning of my joker ark i swear to god i hate byond

Djiq committed:
obsidian

Marmio64 committed:
give tarstone a sprite

Marmio64 committed:
Merge branch 'pr/14802' into give-tarstone-gem-a-sprite

Djiq committed:
Merge pull request #40 from Marmio64/give-tarstone-gem-a-sprite

give tarstone a sprite

Marmio64 committed:
surrounds gulag in lava

Djiq committed:
Merge pull request #41 from Marmio64/gulag-changes

Gives gulag a thin layer of lava

Marmio64 committed:
start of syndi station

Marmio64 committed:
more stuff

Marmio64 committed:
almostthere

Marmio64 committed:
finished

Marmio64 committed:
fixcryo

Marmio64 committed:
gives the comms agent his chameleon mask

Marmio64 committed:
fix one area

Marmio64 committed:
fix some firedoors

Marmio64 committed:
fix a blastdoor

Marmio64 committed:
fire extinguishers and some berets

Marmio64 committed:
should fix thermomachine directions

Marmio64 committed:
heater starts on

Marmio64 committed:
fixes cryo for real

Marmio64 committed:
fix tcomms cooler direction

Marmio64 committed:
paper and mats

Marmio64 committed:
good monkey cubes

Marmio64 committed:
lieutenant gets glasses

Marmio64 committed:
little bit more self defense stuff

Marmio64 committed:
yup

Marmio64 committed:
pipe fix

Marmio64 committed:
fix nitrogen

Marmio64 committed:
extra misc stuff and food

Marmio64 committed:
welding fuel

Marmio64 committed:
Update jungleland_swamp_syndicatestation.dmm

Djiq committed:
Merge pull request #42 from Marmio64/jungle-syndi-station

Jungle syndi station

Djiq committed:
fixes and makes gulag no longer extremely fucking ugly

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
post-merge fixes

Djiq committed:
disables syndie base till it gets fixed

Marmio64 committed:
may fix it?

Djiq committed:
Merge pull request #43 from Marmio64/fixbase

Fixbase?

Djiq committed:
e

Djiq committed:
Revert "e2"

This reverts commit aaa163d81aae0c43b6888dfa9fe1ef56c99d79b9.

Djiq committed:
e2

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
huh

Djiq committed:
fixes

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
fixes toxin being invisible (no clue how tf that happened)

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
BUNCH OF FIXES

Djiq committed:
demon core stuff

Djiq committed:
fixes aux base

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
fixes whiteship error and syndie station gun turrets being too close

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
reworks toxic buildup status effect to be a chemical reagent so you can treat it with toxin medication easier

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
works?

Djiq committed:
wake up

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
should be fixed now, this compile time issue didnt appear on my local

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
e

Djiq committed:
kuku ;_;

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
would this work?

Djiq committed:
nevermind updating dream maker didnt work

Djiq committed:
e2

Djiq committed:
what

Djiq committed:
bump

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
e

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
fixes compile error

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
adjusts values and fixes skin twisters not being able to be pulled after dying

Djiq committed:
mosquito fix

Djiq committed:
adjusts trident

Djiq committed:
bump???

Djiq committed:
more mosquito fixes

Djiq committed:
fixes some bugs with dryad hearts and makes necropolis chests appear only after defeating elite fauna

Djiq committed:
fixed lmao

Djiq committed:
bump

Djiq committed:
fixes dryad hearts being invisible oops

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
tar mobs delete on death

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
adds more deep toxic pits to jungleland syndie base, and makes acid generally slightly deadlier

Marmio64 committed:
remove tar altar ruin and add tar altar to dying tar temple

Marmio64 committed:
a couple visual changes

Marmio64 committed:
forgot to save

Marmio64 committed:
small fishing area

Marmio64 committed:
proper atmos

Djiq committed:
Merge pull request #45 from Marmio64/fishing

syndicate base fishing pond

Djiq committed:
tar king no longer spams message_admins

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
Merge branch 'jungleland' of https://github.com/EdgeLordExe/Yogstation into jungleland

Djiq committed:
fixes skin twister bug??

Djiq committed:
Merge pull request #44 from Marmio64/arena

remove tar altar ruin and add tar altar to dying tar temple

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
tiny change to lavaland geode

Djiq committed:
more geode fixes (atmos mostly)

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
updates tar assistant and tar enchant

Marmio64 committed:
a few tweaks

Djiq committed:
Merge pull request #46 from Marmio64/ruinimprovements

minor ruin tweaks

Djiq committed:
adds magnus purpura, a new herb in jungleland that allows you to temporarily walk on non-deep sulphuric pits without taking damage

Djiq committed:
makes the sulph put immune trait also work on toxic buildup

Djiq committed:
e2

Djiq committed:
Merge branch 'jungleland' of https://github.com/EdgeLordExe/Yogstation into jungleland

Djiq committed:
adds jungle parallax woo hoo!

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
huh?

Djiq committed:
fixes a couple things

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

ToasterBiome committed:
Merge branch 'master' into jungleland

Marmio64 committed:
new tar mobs

Djiq committed:
Merge pull request #47 from Marmio64/newtarmobs

new tar mobs

Djiq committed:
no more shelter on deep sulphuric pit wah wah

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
Merge branch 'jungleland' of https://github.com/EdgeLordExe/Yogstation into jungleland

Djiq committed:
tar enchant clarification

Djiq committed:
e

Djiq committed:
funkies and tar king much harder now B )

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland

Djiq committed:
alpha elites can no longer spawn from gold extracts

Djiq committed:
Merge remote-tracking branch 'yogstation13/master' into jungleland
